checker/check-request-property-type-changed.go

```diff
@@ -0,0 +1,107 @@
+			continue
+		}
+		for operation, operationItem := range pathItem.OperationsDiff.Modified {
+			if operationItem.RequestBodyDiff == nil ||
+				operationItem.RequestBodyDiff.ContentDiff == nil ||
+				operationItem.RequestBodyDiff.ContentDiff.MediaTypeModified == nil {
+				continue
+			}
+			source := (*operationsSources)[operationItem.Revision]
+
+			modifiedMediaTypes := operationItem.RequestBodyDiff.ContentDiff.MediaTypeModified
+			for mediaType, mediaTypeDiff := range modifiedMediaTypes {
+				if mediaTypeDiff.SchemaDiff != nil {
+					typeDiff := mediaTypeDiff.SchemaDiff.TypeDiff
+					formatDiff := mediaTypeDiff.SchemaDiff.FormatDiff
+					if (typeDiff != nil || formatDiff != nil) && (typeDiff == nil || typeDiff != nil &&
+						!(typeDiff.From == "integer" && typeDiff.To == "number") &&
+						!(typeDiff.To == "string" && mediaType != "application/json" && mediaType != "application/xml")) &&
+						(formatDiff == nil || formatDiff != nil && formatDiff.To != nil && formatDiff.To != "" &&
+							!(mediaTypeDiff.SchemaDiff.Revision.Value.Type == "string" &&
+								(formatDiff.From == "date" && formatDiff.To == "date-time" ||
+									formatDiff.From == "time" && formatDiff.To == "date-time")) &&
+							!(mediaTypeDiff.SchemaDiff.Revision.Value.Type == "number" &&
+								(formatDiff.From == "float" && formatDiff.To == "double")) &&
+							!(mediaTypeDiff.SchemaDiff.Revision.Value.Type == "string" &&
+								(formatDiff.From == "decimal" && formatDiff.To == "uuid")) &&
+							!(mediaTypeDiff.SchemaDiff.Revision.Value.Type == "integer" &&
+								(formatDiff.From == "int32" && formatDiff.To == "int64" ||
+									formatDiff.From == "int32" && formatDiff.To == "bigint" ||
+									formatDiff.From == "int64" && formatDiff.To == "bigint"))) {
+						if typeDiff == nil {
+							typeDiff = &diff.ValueDiff{From: mediaTypeDiff.SchemaDiff.Revision.Value.Type, To: mediaTypeDiff.SchemaDiff.Revision.Value.Type}
+						}
+						if formatDiff == nil {
+							formatDiff = &diff.ValueDiff{From: mediaTypeDiff.SchemaDiff.Revision.Value.Format, To: mediaTypeDiff.SchemaDiff.Revision.Value.Format}
+						}
+						result = append(result, BackwardCompatibilityError{
+							Id:        "request-body-type-changed",
+							Level:     ERR,
+							Text:      fmt.Sprintf(config.i18n("request-body-type-changed"), empty2none(typeDiff.From), empty2none(formatDiff.From), empty2none(typeDiff.To), empty2none(formatDiff.To)),
+							Operation: operation,
+							Path:      path,
+							Source:    source,
+						})
+					}
+				}
+
+				CheckModifiedPropertiesDiff(
+					mediaTypeDiff.SchemaDiff,
+					func(propertyPath string, propertyName string, propertyDiff *diff.SchemaDiff, parent *diff.SchemaDiff) {
+						if propertyDiff.Revision.Value.ReadOnly {
+							return
+						}
+						typeDiff := propertyDiff.TypeDiff
+						formatDiff := propertyDiff.FormatDiff
+
+						if (typeDiff != nil || formatDiff != nil) && (typeDiff == nil || typeDiff != nil &&
```

Comment:
Looks like code duplication.
I recommend to reuse it through a function.

Reply:
Done